docs/guides/writing_stubs.rst

```diff
@@ -106,3 +106,473 @@ particularly with false positives.
 If your package has some particularly complex aspects, you could even consider
 writing dedicated typing tests for tricky definitions. For more details, see
 :ref:`testing`.
+
+Stub Content
+============
+
+This section documents best practices on what elements to include or
+leave out of stub files.
+
+Modules excluded fom stubs
+--------------------------
+
+Not all modules should be included into stubs.
+
+It is recommended to exclude:
+
+1. Implementation details, with `multiprocessing/popen_spawn_win32.py <https://github.com/python/cpython/blob/main/Lib/multiprocessing/popen_spawn_win32.py>`_ as a notable example
+2. Modules that are not supposed to be imported, such as ``__main__.py``
+3. Protected modules that start with a single ``_`` char. However, when needed protected modules can still be added (see :ref:`undocumented-objects` section below)
+
```

Comment:
I went ahead and addressed these future notes, since most of the fixes were fairly straightforward, and I'll be splitting up this PR anyway (which should help with the concerns about it being large/unfocused).